Src/Base/AMReX_BArena.cpp

```diff
@@ -31,6 +34,13 @@ amrex::BArena::alloc (std::size_t _sz)
 	gpu_malloc(&pt, &_sz);
 
     }
+// If combining the OpenMP offloading code with the existing CUDA_UM and
+// AMREX_USE_CUDA code, the tutorials generate the wrong answer.  I do not know
+// why. So until that gets sorted out, the OpenMP offloading code can't re-use
+// any of the existing CUDA code here. This will lead to some duplication (such
+// as these calls to cudaMallocManaged() and cudaFree()).
+#elif (defined(OMP_OFFLOAD))
+    cudaMallocManaged(&pt, _sz);
```

Comment:
Can you test whether this issue is due specifically to the use of CArena by commenting out DEFINES += -DBL_COALESCE_FABS in Make.defs?

Reply:
I tested with `USE_OMP_OFFLOAD=TRUE USE_CUDA=TRUE` and removed the line `DEFINES += -DBL_COALESCE_FABS`. Removing that line makes no difference - it gives the same result as if the line were present (which is good), but the answer is still wrong (which is bad).

Reply:
OK. One thing I realized is that we're still using CUDA Fortran to do cudaMallocManaged which I would independently like to change and do it in C++. Let me try that and see if that makes any difference.